src/core_modules/capture-core/components/ScopeSelector/ScopeSelector.container.js

```diff
@@ -1,152 +1,132 @@
-    if (selectedOrgUnitId) {
-        const orgUnit = organisationUnits[selectedOrgUnitId];
-        return Boolean(orgUnit && orgUnit.id && !lockedSelectorLoads);
+    if (selectedOrgUnitId && selectedOrgUnitName) {
+        return true;
     }
     return !lockedSelectorLoads;
 };
 
-const useUrlQueries = (): { selectedProgramId: string, selectedOrgUnitId: string, pathname: string } =>
-    useSelector(({
-        currentSelections: {
-            programId: selectedProgramId,
-            orgUnitId: selectedOrgUnitId,
-        },
-        router: {
-            location: {
-                query: {
-                    programId: routerProgramId,
-                    orgUnitId: routerOrgUnitId,
-                },
-                pathname,
-            },
-        },
-    }) =>
-        ({
-            selectedProgramId: routerProgramId || selectedProgramId,
-            selectedOrgUnitId: routerOrgUnitId || selectedOrgUnitId,
-            pathname,
-        }),
-    );
-
-const useComponentLifecycle = () => {
-    const dispatch = useDispatch();
-    const { selectedOrgUnitId } = useUrlQueries();
-    const { pathname } = useLocation();
-    const pageFetchesOrgUnit = !pageFetchesOrgUnitUsingTheOldWay(pathname.substring(1));
-
-    useEffect(() => {
-        pageFetchesOrgUnit && selectedOrgUnitId && dispatch(fetchOrgUnit(selectedOrgUnitId));
-    },
-    [dispatch, selectedOrgUnitId, pageFetchesOrgUnit]);
-};
-
 export const ScopeSelector: ComponentType<OwnProps> =
   ({
       customActionsOnProgramIdReset = [],
       customActionsOnOrgUnitIdReset = [],
-      pageToPush = '',
       isUserInteractionInProgress = false,
+      selectedProgramId,
+      selectedOrgUnitId,
+      onSetProgramId,
+      onSetOrgUnit,
+      onResetProgramId,
+      onResetOrgUnitId,
+      children,
   }) => {
       const dispatch = useDispatch();
+      const { reset } = useReset();
+      const { resetProgramId } = useResetProgramId();
+      const { referchOrganizationUnit, displayName: selectedOrgUnitName } = useOrganizationUnit();
+      const [selectedOrgUnit, setSelectedOrgUnit] = useState({ name: selectedOrgUnitName, id: selectedOrgUnitId });
 
-      const dispatchOnSetOrgUnit = useCallback(
-          (id: string, orgUnit: Object) => {
-              dispatch(setOrgUnitFromLockedSelector(id, orgUnit, pageToPush));
-          },
-          [pageToPush, dispatch]);
+      useEffect(() => {
+          const missName = !selectedOrgUnit.name;
+          const hasDifferentId = selectedOrgUnit.id !== selectedOrgUnitId;
 
-      const dispatchOnSetProgramId = useCallback(
-          (id: string) => {
-              dispatch(setProgramIdFromLockedSelector(id, pageToPush));
-          },
-          [pageToPush, dispatch]);
+          selectedOrgUnitId && (hasDifferentId || missName) && referchOrganizationUnit(selectedOrgUnitId);
+      },
+      [selectedOrgUnitId]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I might miss something here, but: I think we can safely add the missing dependencies if we wrap `referchOrganizationUnit` in a `useCallback` (https://github.com/dhis2/capture-app/pull/1878/files#diff-ace3d81a85eaec0b1205ec160385ad8ef70fcb223b001aca7f6fc36911e007a1R24)